packages/hash/docker/realtime/prod/Dockerfile

```diff
@@ -27,6 +29,10 @@ WORKDIR /app
 RUN groupadd -g 999 appuser \
   && useradd -r -u 999 -g appuser appuser
 
+USER root
+RUN chown appuser:appuser /app
+
 USER appuser
+ENV NODE_ENV production
 
-CMD ["yarn", "workspace","@hashintel/hash-realtime", "start"]
+CMD ["node", "--max-old-space-size=2048", "./node_modules/.bin/ts-node", "--transpile-only", "./packages/hash/realtime/src/index.ts"]
```

Comment:
See https://stackoverflow.com/a/65789609/3742885 (and linked github PR in comments) for more info on why we do transpile only

Reply:
We could capture this in the README

Reply:
Hmmm not sure if we want to run the app differently depending on whether we are inside a container or not. If we do need `--max-old-space-size=2048`, We should probably include it into the `start` script (or even the `exe` script).

Not sure why the app would consume different amount of RAM based on the Docker presence.

Reply:
I just looked further into this, and according to [this](https://nodejs.org/api/cli.html#--max-old-space-sizesize-in-megabytes) we should be using 1536 MB. Our Fargate tasks are set to 2GB ram. But i definitely do not thing this infrastructure detail should be bubbled up to the container.

> We should probably include it into the start script (or even the exe script).
The API package json defines the `start` script to use this setting already. That is why I added it in the CMD params. Do you think we should omit it here?

Reply:
TLDR: I suggest including all flags inside scripts, if needed. `Dockerfile` should not include a Docker-specific entry point for the app.

Reply:
The last stage of the docker containers do not have yarn (which I think is on purpose?) how much would using the yarn scripts influence container image size

Reply:
Yarn is quite small (it’s just a global npm package). I think the reason why we don't have it is because we use:

```
FROM ubuntu:20.04

COPY --from=builder /app /app
COPY --from=builder /usr/local/bin/node /usr/local/bin/node
```

If we go for `FROM node:16` instead or copy yarn binary, we should get access to `yarn ...` commands.

Reply:
When we change what `yarn ... start` is doing, we should not need to open the `Dockerfile` and change it too. The whole purpose of `package.json` script is to abstract away the implementation detail of an entry point and make it universal.

Reply:
If you use something like `node:16.14.2-alpine`, then `yarn` is included, yes 👍

Reply:
![image](https://user-images.githubusercontent.com/6988668/160454912-9741cc8e-d3b6-48a2-9054-cfdca266d331.png)
Not a bad dip in size

Reply:
Okay should be pushed now. Thanks for the pointers!